### PR TITLE
FEAT: Add windows version

### PR DESCRIPTION
**Pull request by MementoRC**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by MementoRC** [comment]
@xhochy At your convenience, could you please review the addition of windows to this recipe? It uses a binary distribution from upstream to build itself from source. Once it is merged, i will change the recipe to use conda installed version for bootstrap, instead of the hard-coded binary - The idea is to replicate this for all the supported arch

@Tobias-Fischer @wolfv This is ready for your review review

**Comment by Tobias-Fischer** [comment]
One comment - do we really need zstd in build?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -2,40 +2,86 @@
...
+    - MSI_FILE=sbcl-{{ win_64_version }}-x86-64-windows-binary.msi  # [win and x86_64]
 
 requirements:
   build:
-    - {{ stdlib("c") }}
-    - make  # [unix]
+    - {{ compiler('c') }}
+    - {{ stdlib('c') }}
+    - make
+    - zstd
```
Comment (xhochy) [inline recipe/meta.yaml:44]: This should probably only be in host. If not, add a comment.
```suggestion
```

File: recipe/bld.bat [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+set "INSTALL_ROOT=%SRC_DIR%\_bootstrap\PFiles\Steel Bank Common Lisp"
+copy "%INSTALL_ROOT%\sbcl.exe" "%INSTALL_ROOT%\sbcl" > nul
+if %ERRORLEVEL% neq 0 exit /b %ERRORLEVEL%
+
+set "SBCL_HOME=%INSTALL_ROOT%"
+set "PATH=%SBCL_HOME%;%PATH%"
+
+cd %SRC_DIR%\sbcl-source
+  set "PATH=%BUILD_PREFIX%\Library\mingw-w64\bin;%PATH%"
+  set "CC=gcc"
```
Comment (xhochy) [inline recipe/bld.bat:16]: If you really use `gcc` as the compiler, you should be using a different `stdlib` and `compiler` function (i.e. `m2w64_c`) in the meta.yml for windows.

**Review by xhochy — COMMENTED** [review]
In general, it is fine to use an external binary for bootstrapping. We only prefer to have built stuff from source in the final artefact.

File: recipe/bld.bat (hunk 2 above)
Comment (MementoRC) [reply recipe/bld.bat:16]: Yes, `gcc` is actually hard-coded in one of the config files. I used  that compiler initially, but it failed some compilation. Let me give it another try

**Comment by MementoRC** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (MementoRC) [reply recipe/meta.yaml]: SBCL needs zstd in order to save compressed core (in the INSTALL). I opt to add a comment. Note that we could remove this feature instead

File: recipe/bld.bat (hunk 2 above)
Comment (MementoRC) [reply recipe/bld.bat:16]: It seems to me that when I don't specify `m2w64_c` the `gcc` that is used is `12.x` (https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=933100&view=logs&j=a70f640f-cc53-5cd3-6cdc-236a1aa90802&t=6119ccbe-9301-594f-7c27-f792b80a7fcc&l=991)
How to specify this more recent gcc version?

Comment (xhochy) [reply recipe/bld.bat:16]: I actually don't know where this gcc is coming from.

Comment (MementoRC) [reply recipe/bld.bat:16]: I tried to make the recipe depend on the `m2w64-experimental` using `channel_sources` and `channel-targets`, but it does not seem to work. It feels like we are at a dead-end?

**Comment by MementoRC** [comment]
@conda-forge-admin, please rerender

**Comment by MementoRC** [comment]
@xhochy Would this be acceptable?

**Comment by xhochy** [comment]
It's OK, but I can give no good answer here. This may be quite fragile, so let the other maintainers comment here how they feel.

**Comment by MementoRC** [comment]
Sure, it does feel like the more reasonable approach would be to await for the update of `m2w64`, is this what is underway with the `experimental` label? This PR can certainly be put on hold until then

File: recipe/conda_build_config.yaml [hunk 3]
```diff
@@ -2,3 +2,7 @@ c_stdlib_version:  # [unix]
   - '2.28'  # [linux and x86_64]
 docker_image:
   - quay.io/condaforge/linux-anvil-alma-x86_64:8  # [linux and x86_64]
+channel_sources:  # [win]
+  - conda-forge/label/m2w64-experimental,conda-forge  # [win]
+channel_targets:  # [win]
+  - conda-forge m2w64-experimental  # [win]
```
Comment (Tobias-Fischer) [inline recipe/conda_build_config.yaml:8]: I am not sure whether the package needs this `channel_targets` - is the package incompatible with other Windows packages that are not compiled with `m2w64-experimental`? If so, it is basically useless; if not, then we can put it in the main conda-forge channel, can't we?

/cc @xhochy @traversaro

**Comment by Tobias-Fischer** [comment]
Overall this looks good to me; just added one comment. @wolfv if you want to take a look as well, that'd be great.

File: recipe/conda_build_config.yaml (hunk 3 above)
Comment (xhochy) [reply recipe/conda_build_config.yaml:8]: I think it is fine to be put on main. The special channel is only needed to pull the newer `m2w64` compilers.

**Comment by Tobias-Fischer** [comment]
@conda-forge-admin  please rerender

**Pull request merged**